src/cpp/core/system/PosixSystem.cpp

```diff
@@ -1218,11 +1224,39 @@ Error processInfo(const std::string& process, std::vector<ProcessInfo>* pInfo)
                continue;
             }
 
+            // read the stat fields for other relevant process infostd::string str;
+            std::string statStr;
+            error = core::readStringFromFile(statFile, &statStr);
+            if (error)
+            {
+               LOG_ERROR(error);
+               continue;
+            }
+
+            std::vector<std::string> statFields;
+            boost::algorithm::split(statFields, statStr,
+                                    boost::is_any_of(" "),
+                                    boost::algorithm::token_compress_on);
+
+            // get process parent id
+            std::string ppidStr = statFields[3];
```

Comment:
This will throw if we somehow read fewer fields than expected so should be guarded.